server/service/schedule/schedule.go

```diff
@@ -468,22 +532,17 @@ func (s *Schedule) getLockName() string {
 	return name
 }
 
-// getRemainingInterval returns the interval minus the remainder of dividing the time since state by
-// the interval
-func getRemainingInterval(start time.Time, interval time.Duration) time.Duration {
-	if interval == 0 {
-		return 0
-	}
-	return interval - (time.Since(start) % interval)
-}
-
-// clearTickerChannel performs a non-blocking select on the ticker channel
-func clearTickerChannel(ticker *time.Ticker) {
-	select {
-	case <-ticker.C:
-		// pull from ticker channel
-	default:
-		// ok
+// clearScheduleChannels performs a non-blocking select on the ticker and trigger channel
+func clearScheduleChannels(trigger chan struct{}, ticker <-chan time.Time) {
```

Comment:
Hm, this function feels odd to me... let's make sure to discuss when we get on a call.